scanner/src/main/kotlin/provenance/PackageProvenanceResolver.kt

```diff
@@ -220,7 +220,19 @@ class DefaultPackageProvenanceResolver(
 
             revisionCandidates.forEachIndexed { index, revision ->
                 logger.info { "Trying revision candidate '$revision' (${index + 1} of ${revisionCandidates.size})." }
-                val result = vcs.updateWorkingTree(workingTree, revision, pkg.vcsProcessed.path, recursive = false)
+                val result = vcs.updateWorkingTree(workingTree, revision, recursive = false)
+
+                if (pkg.vcsProcessed.path.isNotBlank() &&
+                    !workingTree.workingDir.resolve(pkg.vcsProcessed.path).exists()
+                ) {
+                    logger.info {
+                        "Discarding revision '$revision' because the requested VCS path '${pkg.vcsProcessed.path}' " +
+                                "does not exist."
+                    }
```

Comment:
Sorry, for not noticing this earlier. I just though about the use case where a user
sets a wrong (non-existing) VCS path in a curation. I'm not certain IIRC, but I think previously this lead to an ORT issue indicating that the VCS path does not exist. As of this change, the problem would only be logged, while the actual failure would be: "Cannot determine a revision to checkout". 

In another (more edge case) scenario, where multiple candidates match (by tag name) I'm not sure if using the path for discarding is a good idea.

Would it make sense to not use the path for discarding and do the following instead?
1. Determine the revision to checkout agnostic of the path
2. Report an OrtIssue if the VCS path does not exist in the guessed revision

Reply:
That's a good point, I will make a separate PR to improve the error reporting to catch also other errors which are currently swallowed.

Reply:
So, do you agree that the revision should not be discarded based on the path?

Reply:
No, I think it should be discarded, better fail early than fail in the scanner step when we already know that the path does not exist here. What I want to improve is the error message so that the user knows why no revision was found, this will end up in an `OrtIssue`.

Reply:
I'm not 100% convinced about including the path in matching candidates, but also needs not be discussed in scope of this PR.